Firmware/MotorControl/mechanical_brake.cpp

```diff
@@ -0,0 +1,22 @@
+#include <odrive_main.h>
+
+bool MechanicalBrake::apply_config() {
+    config_.parent = this;
+    set_enabled(config_.enabled);
+    return true;
+}
+
+void MechanicalBrake::engage() {
+    get_gpio(config_.gpio_num).write(config_.is_active_low ? 0 : 1);
+}
+
+void MechanicalBrake::release() {
+    get_gpio(config_.gpio_num).write(config_.is_active_low ? 1 : 0);
+}
+
+void MechanicalBrake::set_enabled(bool enable) { 
+    if (enable) {
+        // We need this flag to alert the system to the configuration on boot
+	    odrv.config_.gpio_modes[config_.gpio_num] = ODriveIntf::GPIO_MODE_MECH_BRAKE;
```

Comment:
I think it's better not to automatically change the pin config here. If every enabled component changes pins to itself then we're kinda back where we were before where it's undefined, given a certain configuration, which component "wins" the GPIO.

Removing this function also allows us to remove some more code: the `Config_t::set_enabled()` and `Config_t::set_gpio_num()` hooks, consequently the `Config_t::parent` field and, further again, the `MechanicalBrake::apply_config()` function.